src/main/java/org/gooru/insights/services/CSVFileWriterServiceImpl.java

```diff
@@ -63,7 +66,7 @@ public void generateCSVReport(Set<String> headerKeys, List<Map<String, Object>>
 			StringBuilder rowLine = new StringBuilder(); 
 			for (Map<String, Object> row : rowList) {
 				for(String headerKey : headerKeys) {
-					String key = StringUtils.isBlank(row.get(headerKey).toString()) ? APIConstants.NOT_APPLICABLE : row.get(headerKey).toString();
+					String key = row.get(headerKey) == null || StringUtils.isBlank(row.get(headerKey).toString()) ? APIConstants.NOT_APPLICABLE : row.get(headerKey).toString();
```

Comment:
@ManikandanV 
Can you remove .toString() casting?. If you need String value alone then can add Empty String there? so that we can avoid conversion time?